### ISSUE TITLE
DB specific error messages and codes 

### ISSUE THREAD

**Issue by arunkumar413** [knex/knex#5945]
Hi Team,

In documentation I don't see any information related to the database specific error messages and codes. Could you please provide more  information on how to account for the database error messages while using knex.

Thanks,
Arun

**Comment by mercmobily** [comment]
Hello Arun, thanks for the suggestion.

This is a documentation gap; Knex surfaces underlying driver errors without normalising vendor-specific codes. We should add a docs section clarifying this and linking to driver references.

Knex executes the compiled SQL by delegating to the selected driver and returns the driver’s error object unchanged. For example, PostgreSQL builds a `queryConfig` with `text` and `values` and forwards it to `pg` (lib/dialects/postgres/index.js:170-209); MySQL creates a connection and calls `connection.query` (lib/dialects/mysql/index.js:63-80 and 100-144). The core client does not remap codes; errors propagate from the driver (lib/client.js:308-339), which is why you’ll see `code`/`errno`/`sqlState`/`constraint` fields that are specific to each database.

Typical patterns by driver:

- PostgreSQL (`pg`): `error.code` (SQLSTATE), `error.constraint`, `error.detail`, `error.schema`, `error.table`.
- MySQL/MariaDB (`mysql2`): `error.code` (string like `ER_DUP_ENTRY`), `error.errno` (numeric), `error.sqlState`.
- MSSQL (`mssql`): `error.number`, `error.code`, `error.state`, `error.class`.
- Oracle (`oracledb`): `error.errorNum`, `error.offset`, `error.message`.

Example handling across drivers:

```
try {
  await knex('users').insert({ id: 1, email: 'duplicate@example.com' });
} catch (err) {
  // Inspect driver-specific fields
  if (err.code === '23505' || err.code === 'ER_DUP_ENTRY' || err.number === 2627) {
    // unique violation across pg/mysql/mssql
  }
  // Log full error for diagnosis
}
```

In conclusion, this is by design; Knex intentionally preserves driver error objects. I propose adding a documentation section that enumerates common error fields per driver and links to official references, so users can reliably branch on vendor-specific metadata.
